sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -368,3 +387,7 @@ public void visitValue(PValue value, Node producer) {
     return visitor.boundedness == IsBounded.UNBOUNDED;
   }
 }
+
+class ClassForNull {
```

Comment:
I don't want to use synchronization in ParDo is because that's anti pattern. Using ParDo means we do something in parallel, and synchronization in this problem means I will essentially make `processElement` run like in a single thread mode. Also, it is not clear to me at all that how to make synchronization cross bundles, and clearly synchronization makes too much assumptions on direct runner.

Reply:
`ConcurrentLinkedQueue`, instead, only assumes that direct runner runs this primitive by Java in such a way that every bundle runs in different threads can access the same memory location.

Reply:
Another reason is performance. Synchronization will hurt the performance for sure. I don't have number for performance regression though.